### PR TITLE
Add per api ingress support

### PR DESCRIPTION
**Pull request by Uvindu96**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: apim-operator/pkg/controller/api/api_controller.go [hunk 1]
```diff
@@ -845,7 +845,7 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
 					reqLogger.Info("Operator mode is set to " + operatorMode)
 					if operatorMode == ingressMode {
 						ingErr := createorUpdateMgwIngressResource(r, instance, userNameSpace, int32(httpPortVal),
-							int32(httpsPortVal), apiBasePaths, controlIngressConf)
+							int32(httpsPortVal), apiBasePaths, controlIngressConf,owner)
```
Comment (pubudu538) [inline apim-operator/pkg/controller/api/api_controller.go]: Formatting issue

Comment (Uvindu96) [reply apim-operator/pkg/controller/api/api_controller.go]: updated

File: apim-operator/pkg/controller/api/api_controller.go [hunk 2]
```diff
@@ -1879,16 +1879,16 @@ func createMgwLBService(r *ReconcileAPI, cr *wso2v1alpha1.API, nameSpace string,
 // Creating a LB balancer service to expose mgw
 // Supports for multiple apiBasePaths when there are multiple swaggers for one API CRD
 func createorUpdateMgwIngressResource(r *ReconcileAPI, cr *wso2v1alpha1.API, nameSpace string, httpPortVal int32,
-	httpsPortVal int32, apiBasePaths []string, controllerConfig *corev1.ConfigMap) error {
+	httpsPortVal int32, apiBasePaths []string, controllerConfig *corev1.ConfigMap,owner []metav1.OwnerReference) error {
```
Comment (pubudu538) [inline apim-operator/pkg/controller/api/api_controller.go]: Formatting issues.

Comment (Uvindu96) [reply apim-operator/pkg/controller/api/api_controller.go]: updated

**Pull request merged**